compiler/rustc_errors/messages.ftl

```diff
@@ -1,3 +1,9 @@
+errors_expected_lifetime_parameter_plural =
+    expected lifetime parameters
+
+errors_expected_lifetime_parameter_single =
+    expected lifetime parameter
```

Comment:
The way you set this up by having two separate Fluent messages to support grammatical numbers is not quite right.
For example, (A) there are languages that don't mark the plural at all and (B) ones that not only have a singular and a plural but also a *dual* for marking nouns with a quantity of exactly two. For (A), this setup requires duplication and for (B) it's outright impossible to translate this Fluent message.

Please store the exact amount in `ExpectedLifetimeParameter`, e.g. `pub(crate) count: usize` and adjust the Fluent file for en-US:

```suggestion
errors_expected_lifetime_parameters =
    expected lifetime {$count ->
        [one] parameter
        *[other] parameters
    }
```

You can read more about translatable diagnostics [here](https://rustc-dev-guide.rust-lang.org/diagnostics/translation.html).

Reply:
It really looks better now 👍